### DISCUSSION TITLE
requestLayout() improperly called

### DISCUSSION THREAD

**Discussion by russnash** [dotnet/maui#6512] [Q&A]
Just wondering if anyone else is seeing floods of this while their app is running?  Just updated to RC2 and my app has slowed to a crawl while endlessly spitting out these messages:

`[View] requestLayout() improperly called by crc64e1fb321c08285b90.FrameRenderer{818c397 V.E...... ......ID 14,14-1067,667 #6} during layout: running second layout pass
[View] requestLayout() improperly called by crc6452ffdc5b34af3a0f.LayoutViewGroup{e455a2 V.E...... ........ 53,53-1002,601} during second layout pass: posting in next frame
[View] requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{771ab69 V.ED..... ........ 0,0-949,103} during second layout pass: posting in next frame
[View] requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{47d43dd V.ED..... ........ 0,103-949,522} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc64e1fb321c08285b90.FrameRenderer{818c397 V.E...... ......I. 14,14-1067,667 #6} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc64e1fb321c08285b90.FrameRenderer{818c397 V.E...... ......ID 14,14-1067,667 #6} during layout: running second layout pass
[View] requestLayout() improperly called by crc6452ffdc5b34af3a0f.LayoutViewGroup{e455a2 V.E...... ........ 53,53-1002,601} during second layout pass: posting in next frame
[View] requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{771ab69 V.ED..... ........ 0,0-949,103} during second layout pass: posting in next frame
[View] requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{47d43dd V.ED..... ........ 0,103-949,522} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc64e1fb321c08285b90.FrameRenderer{818c397 V.E...... ......I. 14,14-1067,667 #6} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc64e1fb321c08285b90.FrameRenderer{818c397 V.E...... ......ID 14,14-1067,667 #6} during layout: running second layout pass
[View] requestLayout() improperly called by crc6452ffdc5b34af3a0f.LayoutViewGroup{e455a2 V.E...... ........ 53,53-1002,601} during second layout pass: posting in next frame
[View] requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{771ab69 V.ED..... ........ 0,0-949,103} during second layout pass: posting in next frame
[View] requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{47d43dd V.ED..... ........ 0,103-949,522} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc64e1fb321c08285b90.FrameRenderer{818c397 V.E...... ......I. 14,14-1067,667 #6} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc64e1fb321c08285b90.FrameRenderer{818c397 V.E...... ......ID 14,14-1067,667 #6} during layout: running second layout pass
[View] requestLayout() improperly called by crc6452ffdc5b34af3a0f.LayoutViewGroup{e455a2 V.E...... ........ 53,53-1002,601} during second layout pass: posting in next frame
[View] requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{771ab69 V.ED..... ........ 0,0-949,103} during second layout pass: posting in next frame
[View] requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{47d43dd V.ED..... ........ 0,103-949,522} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc64e1fb321c08285b90.FrameRenderer{818c397 V.E...... ......I. 14,14-1067,667 #6} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc64e1fb321c08285b90.FrameRenderer{818c397 V.E...... ......ID 14,14-1067,667 #6} during layout: running second layout pass
[View] requestLayout() improperly called by crc6452ffdc5b34af3a0f.LayoutViewGroup{e455a2 V.E...... ......ID 53,53-1002,601} during second layout pass: posting in next frame
`

**Comment by EhhMaybeLater** [comment]
I am getting something similar. It seems to happen after await Navigation.PushAsync() for me. 
17.2 Preview 5.0

**Reply by russnash** [reply]
Switching my code to use border instead of frame rectified this issue for me.

**Reply by hartez** [reply]
Created issue #6755 - we'll take a look at this.

**Reply by gfmoore** [reply]
Yes I'm getting something similar. It is a page with a CollectionView containing labels and imagebuttons and a button at bottom. I get the error message with the button. I replaced the button in the list with an image button, but these do not give the error unlike the previous button, and this is bound to a ViewModel

I only get the error on the first button press after that it's ok
Here's the error list
```
[View] requestLayout() improperly called by crc640ec207abc449b2ca.ContainerView{2b3b435 V.E...... ......ID 0,0-743,166} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc6452ffdc5b34af3a0f.LayoutViewGroup{d82e23b V.E...... ......ID 0,0-743,166} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc640ec207abc449b2ca.ContainerView{1b264d2 V.E...... ......ID 0,0-743,166} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc6452ffdc5b34af3a0f.LayoutViewGroup{791c4a0 V.E...... ......ID 0,0-743,166} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc640ec207abc449b2ca.ContainerView{5183da V.E...... ......ID 0,0-743,166} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc6452ffdc5b34af3a0f.LayoutViewGroup{2904ee8 V.E...... ......ID 0,0-743,166} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc640ec207abc449b2ca.ContainerView{b168de2 V.E...... ......ID 0,0-743,166} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc6452ffdc5b34af3a0f.LayoutViewGroup{e7a3c30 V.E...... ......ID 0,0-743,166} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc640ec207abc449b2ca.ContainerView{59ca2ea V.E...... ......ID 0,0-743,166} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc6452ffdc5b34af3a0f.LayoutViewGroup{72fac78 V.E...... ......ID 0,0-743,166} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc640ec207abc449b2ca.ContainerView{dcae2f2 V.E...... ......ID 0,0-743,166} during second layout pass: posting in next frame
[View] requestLayout() improperly called by crc6452ffdc5b34af3a0f.LayoutViewGroup{9bcbfc0 V.E...... ......ID 0,0-743,166} during second layout pass: posting in next frame

```

Here's the xaml (I can add the ViewModel if you wish.
```
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:local="clr-namespace:TSDZ2Monitor.ViewModels"
             x:Class="TSDZ2Monitor.Pages.BluetoothPage"
             Title="Bluetooth Page">

  <ContentPage.BindingContext>
    <local:BluetoothPeripheralsViewModel />
  </ContentPage.BindingContext>


  <StackLayout>
    <Label Text="Bluetooth LE Connections" />
    <Label Text="{Binding Test}" />

    <CollectionView
      ItemsSource="{Binding BluetoothPeripherals}"
      EmptyView="No data available"
      VerticalOptions="StartAndExpand"
      SelectionMode="Single"
      ItemsLayout="VerticalList"
      Margin="0, 0, 0, 0"
      HeightRequest="250">
      <CollectionView.ItemTemplate>
        <DataTemplate>
          <Grid
             ColumnDefinitions="*,*,auto">

            <Label Grid.Column="0" 
                   Text="{Binding Name}"
                   FontSize="20"
                   Padding="0, 5, 0, 0"
                   TextColor="Yellow"/>
            
            <Label Grid.Column="1" 
                   Text="{Binding Id}" 
                   FontSize="20"
                   Padding="0, 5, 0, 0"
                   TextColor="Yellow"/>

            <ImageButton 
                    Grid.Column="2" 
                    HorizontalOptions="End"  
                    Source="deletebin.svg" 
                    Scale="0.7" 
                    BackgroundColor="Transparent"
                    Command="{Binding Source={RelativeSource AncestorType={x:Type local:BluetoothPeripheralsViewModel}}, Path=DeleteBLEItemCommand}"
                    CommandParameter="{Binding .}"/>

          </Grid>
        </DataTemplate>
      </CollectionView.ItemTemplate>
    </CollectionView>

    <Button Text="{Binding ScanButtonText}"
            WidthRequest="180"
            HeightRequest="55"
            BackgroundColor="LightGreen"
            FontSize="20"
            Command="{Binding ScanBLEPeripheralsCommand}"/>
    <Label Text="{Binding ScanResults}"
           FontSize="20"
           TextColor="LightGreen"/>
  </StackLayout>
</ContentPage>
```

**Reply by hartez** [reply]
@gfmoore This sounds like a different (but similar) issue. The original was specific to Frame. I've created a new issue (#7902) so we can investigate.
